site/js/modules/webui/course.js

```diff
@@ -81,94 +81,79 @@ function handlerEmail(path, params, context, container) {
-                        <div class="input-field">
-                            <label for="email-to">To</label>
-                            <input type="text" id="email-to" name="to" />
-                        </div>
-                        <div class="input-field">
-                            <label for="email-cc">CC</label>
-                            <input type="text" id="email-cc" name="cc" />
-                        </div>
-                        <div class="input-field">
-                            <label for="email-bcc">BCC</label>
-                            <input type="text" id="email-bcc" name="bcc" />
-                        </div>
-                        <div class="input-field">
-                            <label for="email-subject">Subject</label>
-                            <input type="text" id="email-subject" name="subject" required />
-                        </div>
-                        <div class="input-field">
-                            <label for="email-body">Content</label>
-                            <textarea id="email-body" name="body" rows="10"></textarea>
-                        </div>
-                        <div class="checkbox-field">
-                            <input type="checkbox" id="dry-run" name="dry-run" />
-                            <label for="dry-run">Send as Dry Run</label>
-                        </div>
-                        <div class="checkbox-field">
-                            <input type="checkbox" id="html-format" name="html-format" />
-                            <label for="html-format">Content is in HTML</label>
-                        </div>
-                    </fieldset>
-                </div>
-                <button class="send-email">Send Email</button>
-                <div class="results-area"></div>
-            </div>
-        </div>
+    let description = `
+        Separate recipient values with commas.
+        For valid recipient values reference
+        this <a href=${COURSE_USER_REFERENCE_DOC_LINK} target="_blank">documentation</a>.
     `;
 
-    document.querySelector('button.send-email').addEventListener('click', function(event) {
-        Routing.loadingStart(container.querySelector(".results-area"), false);
+    let inputFields = [
+        new Input.FieldType(context, 'to', 'To', {
+            extractInputFunc: extractRecipients,
+        }),
+        new Input.FieldType(context, 'cc', 'CC', {
+            extractInputFunc: extractRecipients,
+        }),
+        new Input.FieldType(context, 'bcc', 'BCC', {
+            extractInputFunc: extractRecipients,
+        }),
+        new Input.FieldType(context, 'subject', 'Subject', {
+            required: true,
+        }),
+        new Input.FieldType(context, 'content', 'Content', {
+            type: Input.INPUT_TYPE_TEXTAREA,
+            additionalAttributes: 'rows="10"',
+        }),
+        new Input.FieldType(context, 'dry-run', 'Send as Dry Run', {
```

Comment:
Optional

You may want to use a key like `dryrun` so you can use dot notation to refer to it later, .e.g, `inputParams.dryrun`.